activerecord/test/cases/relations_test.rb

```diff
@@ -2050,6 +2050,32 @@ def test_relation_with_private_kernel_method
     assert_equal [accounts(:signals37)], sub_accounts.available
   end
 
+  def test_where_with_take_memoization
+    5.times do |idx|
+      Post.create!(title: idx.to_s, body: idx.to_s)
+    end
+
+    posts = Post.all
```

Comment:
Let's put an `order(:title)` on here, otherwise the test will fail sporadically when the database happens to return "3" in the first `take`

Reply:
There is only one Post in the DB with title '3'. 
Will this still be a problem?

Reply:
Oh, sorry, I was concerned that "3" might get [legitimately] chosen as `first_post`, and then the `assert_not_equal` would fail -- but it's comparing `object_id`, so it will actually be fine even if they both contain the same row. :+1: